operators/src/processing/vector_join/equi_data_join.rs

```diff
@@ -0,0 +1,401 @@
+    right_column_prefix: String,
+}
+
+impl<G> EquiLeftJoinProcessor<G>
+where
+    G: Geometry + ArrowTyped + Sync + Send + 'static,
+    for<'g> FeatureCollection<G>: IntoGeometryIterator<'g>,
+    for<'g> <FeatureCollection<G> as IntoGeometryIterator<'g>>::GeometryType: Into<G>,
+    FeatureCollectionRowBuilder<G>: GeoFeatureCollectionRowBuilder<G>,
+{
+    pub fn new(
+        left_processor: Box<dyn VectorQueryProcessor<VectorType = FeatureCollection<G>>>,
+        right_processor: Box<dyn VectorQueryProcessor<VectorType = DataCollection>>,
+        left_column: String,
+        right_column: String,
+        right_prefix: String,
+    ) -> Self {
+        Self {
+            vector_type: PhantomData,
+            left_processor,
+            right_processor,
+            left_column,
+            right_column,
+            right_column_prefix: right_prefix,
+        }
+    }
+
+    fn join<'c>(
+        &self,
+        left: &'c FeatureCollection<G>,
+        right: &DataCollection,
+    ) -> Result<FeatureCollection<G>> {
+        let right = self.fix_columns(right, &left.column_names().collect())?;
+
+        let left_join_column = left.data(&self.left_column)?;
+        let right_join_column = right.data(&self.right_column)?;
+
+        // TODO: use macro
+        match (left_join_column, right_join_column) {
+            (FeatureDataRef::Number(c1), FeatureDataRef::Number(c2)) => {
+                Self::compute_join(&c1, c2, left, &right)
+            }
+            (FeatureDataRef::Decimal(c1), FeatureDataRef::Decimal(c2)) => {
+                Self::compute_join(&c1, c2, left, &right)
+            }
+            (FeatureDataRef::Text(c1), FeatureDataRef::Text(c2)) => {
+                Self::compute_join(&c1, c2, left, &right)
+            }
+            (FeatureDataRef::Categorical(c1), FeatureDataRef::Categorical(c2)) => {
+                Self::compute_join(&c1, c2, left, &right)
+            }
+            (left, right) => Err(Error::ColumnTypeMismatch {
+                left: (&left).into(),
+                right: (&right).into(),
+            }),
+        }
+    }
+
+    fn compute_join<'i, D1, D2, T>(
+        left_join_values: &D1,
+        right_join_values: D2,
+        left: &FeatureCollection<G>,
+        right: &DataCollection,
+    ) -> Result<FeatureCollection<G>>
+    where
+        D1: DataRef<'i, T> + 'i,
+        D2: DataRef<'i, T> + 'i,
+        T: 'i + PartialEq,
+    {
+        let matches_iter = left_join_values
+            .as_ref()
+            .iter()
+            .zip(left.time_intervals())
+            .map(move |(left_value, left_time_interval)| {
+                right_join_values
+                    .as_ref()
+                    .iter()
+                    .zip(right.time_intervals())
+                    .map(|(right_value, right_time_interval)| {
+                        (left_value == right_value)
+                            && left_time_interval.intersects(right_time_interval)
```

Comment:
would maybe be faster to collect the intersection as an Option because your are doint work twice if you first check for intersection and then compute it later on

Reply:
4794702